Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1372,8 +1374,11 @@ def visit_WithStatNode(self, node):
                     name, value = directive
                     if name in ('nogil', 'gil'):
                         # special case: in pure mode, "with nogil" spells "with cython.nogil"
-                        node = Nodes.GILStatNode(node.pos, state = name, body = node.body)
+                        condition = node.manager.args[0] if isinstance(node.manager, ExprNodes.CallNode) else None
```

Comment:
Do we make sure somewhere that there is only an `args[0]` and not an `args[1]` etc.? Or even keywords?

Reply:
Fixed.